From 641104c63acdcb2e3e2f6a68a1c3d0be7d12f80d Mon Sep 17 00:00:00 2001
From: Muhammad Alfi Syahrin <malfisya.dev@hotmail.com>
Date: Fri, 26 Jan 2024 16:16:03 +0700
Subject: [PATCH] Fix mime-info file

- Rename 'icon' to 'generic-icon' in the mime-info file
- Remove mime types for img/raw/rom files
---
 data/fuse.xml.in | 87 ++++++++++++++++++++----------------------------
 1 file changed, 37 insertions(+), 50 deletions(-)

diff --git a/data/fuse.xml.in b/data/fuse.xml.in
index 1d99f91..14e566a 100644
--- a/data/fuse.xml.in
+++ b/data/fuse.xml.in
@@ -5,17 +5,17 @@
     <magic>
       <match type="string" offset="0" value="Compressed Square Wave\x1a"/>
     </magic>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.csw"/>
   </mime-type>
   <mime-type type="application/x-spectrum-dck">
     <comment>ZX Spectrum cartridge</comment>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.dck"/>
   </mime-type>
   <mime-type type="application/x-spectrum-d80">
     <comment>ZX Spectrum disk</comment>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.d40"/>
     <glob pattern="*.d80"/>
   </mime-type>
@@ -25,7 +25,7 @@
       <match type="string" offset="0" value="MV - CPCEMU Disk-File\x0d\x0aDisk-Info\x0d\x0a"/>
       <match type="string" offset="0" value="EXTENDED CPC DSK File\x0d\x0aDisk-Info\x0d\x0a"/>
     </magic>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.dsk"/>
   </mime-type>
   <mime-type type="application/x-spectrum-fdi">
@@ -33,7 +33,7 @@
     <magic>
       <match type="string" offset="0" value="FDI"/>
     </magic>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.fdi"/>
   </mime-type>
   <mime-type type="application/x-spectrum-hdf">
@@ -41,42 +41,37 @@
     <magic>
       <match type="string" offset="0" value="RS-IDE\x1a"/>
     </magic>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.hdf"/>
   </mime-type>
-  <mime-type type="application/x-spectrum-img">
-    <comment>ZX Spectrum disk</comment>
-    <icon name="application-x-spectrum"/>
-    <glob pattern="*.img"/>
-  </mime-type>
   <mime-type type="application/x-spectrum-ltp">
     <comment>ZX Spectrum tape</comment>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.ltp"/>
   </mime-type>
   <mime-type type="application/x-spectrum-mdr">
     <comment>ZX Spectrum microdrive</comment>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.mdr"/>
   </mime-type>
   <mime-type type="application/x-spectrum-mgt">
     <comment>ZX Spectrum disk</comment>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.mgt"/>
   </mime-type>
   <mime-type type="application/x-spectrum-mgtsnp">
     <comment>ZX Spectrum snapshot</comment>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.mgtsnp"/>
   </mime-type>
   <mime-type type="application/x-spectrum-opd">
     <comment>ZX Spectrum disk</comment>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.opd"/>
   </mime-type>
   <mime-type type="application/x-spectrum-opu">
     <comment>ZX Spectrum disk</comment>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.opu"/>
   </mime-type>
   <mime-type type="application/x-spectrum-pzx">
@@ -84,28 +79,15 @@
     <magic>
       <match type="string" offset="0" value="PZXT"/>
     </magic>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.pzx"/>
   </mime-type>
-  <mime-type type="application/x-spectrum-raw">
-    <comment>ZX Spectrum tape</comment>
-    <magic>
-      <match type="string" offset="0" value="\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0Raw tape sample"/>
-    </magic>
-    <icon name="application-x-spectrum"/>
-    <glob pattern="*.raw"/>
-  </mime-type>
-  <mime-type type="application/x-spectrum-rom">
-    <comment>ZX Spectrum cartridge</comment>
-    <icon name="application-x-spectrum"/>
-    <glob pattern="*.rom"/>
-  </mime-type>
   <mime-type type="application/x-spectrum-rzx">
     <comment>ZX Spectrum action replay</comment>
     <magic>
       <match type="string" offset="0" value="RZX!"/>
     </magic>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.rzx"/>
   </mime-type>
   <mime-type type="application/x-spectrum-sad">
@@ -113,7 +95,7 @@
     <magic>
       <match type="string" offset="0" value="Aley's disk backup"/>
     </magic>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.sad"/>
   </mime-type>
   <mime-type type="application/x-spectrum-scl">
@@ -121,38 +103,43 @@
     <magic>
       <match type="string" offset="0" value="SINCLAIR"/>
     </magic>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.scl"/>
   </mime-type>
   <mime-type type="application/x-spectrum-slt">
     <comment>ZX Spectrum snapshot</comment>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.slt"/>
   </mime-type>
   <mime-type type="application/x-spectrum-sna">
     <comment>ZX Spectrum snapshot</comment>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.sna"/>
     <glob pattern="*.snapshot"/>
   </mime-type>
+  <mime-type type="application/x-spectrum-s">
+    <comment>ZX Spectrum snapshot</comment>
+    <generic-icon name="application-x-spectrum"/>
+    <glob pattern="*.s"/>
+  </mime-type>
   <mime-type type="application/x-spectrum-snp">
     <comment>ZX Spectrum snapshot</comment>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.snp"/>
   </mime-type>
   <mime-type type="application/x-spectrum-sp">
     <comment>ZX Spectrum snapshot</comment>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.sp"/>
   </mime-type>
   <mime-type type="application/x-spectrum-spc">
     <comment>ZX Spectrum tape</comment>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.spc"/>
   </mime-type>
   <mime-type type="application/x-spectrum-sta">
     <comment>ZX Spectrum tape</comment>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.sta"/>
   </mime-type>
   <mime-type type="application/x-spectrum-szx">
@@ -160,7 +147,7 @@
     <magic>
       <match type="string" offset="0" value="ZXST"/>
     </magic>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.szx"/>
     <glob pattern="*.zx-state"/>
   </mime-type>
@@ -170,12 +157,12 @@
       <match type="string" offset="0" value="TD"/>
       <match type="string" offset="0" value="td"/>
     </magic>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.td0"/>
   </mime-type>
   <mime-type type="application/x-spectrum-tap">
     <comment>ZX Spectrum tape</comment>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.tap"/>
   </mime-type>
   <mime-type type="application/x-spectrum-trd">
@@ -183,7 +170,7 @@
     <magic>
       <match type="string" offset="0" value="SINCLAIR"/>
     </magic>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.trd"/>
   </mime-type>
   <mime-type type="application/x-spectrum-tzx">
@@ -191,7 +178,7 @@
     <magic>
       <match type="string" offset="0" value="ZXTape!\x1a"/>
     </magic>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.tzx"/>
   </mime-type>
   <mime-type type="application/x-spectrum-udi">
@@ -199,12 +186,12 @@
     <magic>
       <match type="string" offset="0" value="UDI!"/>
     </magic>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.udi"/>
   </mime-type>
   <mime-type type="application/x-spectrum-z80">
     <comment>ZX Spectrum snapshot</comment>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.z80"/>
   </mime-type>
   <mime-type type="application/x-spectrum-zxs">
@@ -212,7 +199,7 @@
     <magic>
       <match type="string" offset="0" value="SNAP"/>
     </magic>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.zxs"/>
   </mime-type>
   ##LIBSPECTRUM_HAVE_AUDIOFILE_BEGIN##
@@ -228,7 +215,7 @@
   ##LIBSPECTRUM_HAVE_ZLIB_BEGIN##
   <mime-type type="application/x-spectrum-compressed">
     <comment>ZX Spectrum compressed</comment>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.csw.gz"/>
     <glob pattern="*.dck.gz"/>
     <glob pattern="*.mgtsnp.gz"/>
@@ -256,7 +243,7 @@
   ##LIBSPECTRUM_HAVE_BZ2_BEGIN##
   <mime-type type="application/x-spectrum-compressed-bz2">
     <comment>ZX Spectrum compressed</comment>
-    <icon name="application-x-spectrum"/>
+    <generic-icon name="application-x-spectrum"/>
     <glob pattern="*.csw.bz2"/>
     <glob pattern="*.dck.bz2"/>
     <glob pattern="*.mgtsnp.bz2"/>
-- 
2.43.0

